docs/src/testing/gas-etimation.md

```diff
@@ -0,0 +1,34 @@
+# Gas estimation
+
+Transaction gas cost is always important for the users. Developers need to know if their contracts are well-optimised
+and gas-efficient.
+Forge supports gas estimation for each test case. All computations are based on the official
+[Starknet docs](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/fee-mechanism/#how_much_fee_is_charged_high_level_overview).
+
+> ⚠️ In the next releases, gas estimation will be improved to include [onchain data cost](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/fee-mechanism/#_on_chain_data)  ⚠️
+
+## Calculating gas from VM resources
+
+[General case](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/fee-mechanism/#general_case)
+of the docs is focused on gas used by running Cairo code. It is mentioned that the gas cost is connected to
+the most costly component. What does it mean?
+
+Let's assume we have a function which uses 100 `Cairo steps`, 12 `range check builtins` and one `keccak builtin`.
+Table (from the `General case`) has gas cost defined for all builtins. In our case:
+- `Cairo step` - 0.01
+- `range check builtin` - 0.16
+- `keccak builtin` - 20.48
+
+Multiplication of those values gives us a gas cost for each component:
+- `Cairo steps` - 100 * 0.01 = 1
+- `range check builtins` - 12 * 0.16 = 1.92
+- `keccak builtin` - 1 * 20.48 = 20.48
+
+We should remember that only the most expensive factor will be taken into account, so our overall gas cost is `20.48`.
+
+## Displaying estimated gas
```

Comment:
I think this section could be just a part of the first one

Reply:
I will reorder the sections like @cptartur said. Just want to keep displaying separate from the calculations